modules/gmc-agent/src/main/java/org/apache/ignite/agent/Agent.java

```diff
@@ -0,0 +1,403 @@
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.agent.dto.action.Request;
+import org.apache.ignite.agent.service.ActionService;
+import org.apache.ignite.agent.service.CacheService;
+import org.apache.ignite.agent.service.ClusterService;
+import org.apache.ignite.agent.service.config.NodeConfigurationExporter;
+import org.apache.ignite.agent.service.config.NodeConfigurationService;
+import org.apache.ignite.agent.service.event.EventsExporter;
+import org.apache.ignite.agent.service.event.EventsService;
+import org.apache.ignite.agent.service.metrics.MetricExporter;
+import org.apache.ignite.agent.service.metrics.MetricsService;
+import org.apache.ignite.agent.service.tracing.GmcSpanExporter;
+import org.apache.ignite.agent.service.tracing.TracingService;
+import org.apache.ignite.cluster.ClusterNode;
+import org.apache.ignite.events.DiscoveryEvent;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.cluster.IgniteClusterImpl;
+import org.apache.ignite.internal.managers.discovery.DiscoCache;
+import org.apache.ignite.internal.processors.gmc.ManagementConfiguration;
+import org.apache.ignite.internal.processors.gmc.ManagementConsoleProcessor;
+import org.apache.ignite.internal.processors.metastorage.DistributedMetaStorage;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.X;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.eclipse.jetty.io.EofException;
+import org.eclipse.jetty.websocket.api.UpgradeException;
+import org.springframework.messaging.simp.stomp.ConnectionLostException;
+import org.springframework.messaging.simp.stomp.StompCommand;
+import org.springframework.messaging.simp.stomp.StompFrameHandler;
+import org.springframework.messaging.simp.stomp.StompHeaders;
+import org.springframework.messaging.simp.stomp.StompSession;
+import org.springframework.messaging.simp.stomp.StompSessionHandlerAdapter;
+
+import static java.util.Collections.singletonList;
+import static org.apache.ignite.agent.StompDestinationsUtils.buildActionRequestTopic;
+import static org.apache.ignite.agent.StompDestinationsUtils.buildMetricsPullTopic;
+import static org.apache.ignite.agent.utils.AgentUtils.monitoringUri;
+import static org.apache.ignite.agent.utils.AgentUtils.toWsUri;
+import static org.apache.ignite.events.EventType.EVT_NODE_FAILED;
+import static org.apache.ignite.events.EventType.EVT_NODE_LEFT;
+import static org.apache.ignite.events.EventType.EVT_NODE_SEGMENTED;
+
+/**
+ * Management Agent.
+ */
+public class Agent extends ManagementConsoleProcessor {
+    /** GMC configuration meta storage prefix. */
+    private static final String MANAGEMENT_CFG_META_STORAGE_PREFIX = "gmc-cfg";
+
+    /** Discovery event on restart agent. */
+    private static final int[] EVTS_DISCOVERY = new int[] {EVT_NODE_FAILED, EVT_NODE_LEFT, EVT_NODE_SEGMENTED};
```

Comment:
Create issue for programmatic activation for all events that Management Console needed.

Reply:
https://ggsystems.atlassian.net/browse/GG-25518